src/xmldocvisitor.cpp

```diff
@@ -84,6 +95,75 @@ static void visitPreStart(TextStream &t, const char *cmd, bool doCaption,
     visitCaption(parent, children);
     t << "\"";
   }
+  if ((cmd == "dotfile") || (cmd =="dot"))
+  {
+    QCString baseName=name;
```

Comment:
I count 4 copies of the code from lines 100-109. Better to factor out this part.

Reply:
At which commit did you look?

It looks like you looked at the commit:
```
Commit: 723c6eabbb48ce43c203044ca0fde132a0c7249b [723c6ea]
Date: Tuesday, January 11, 2022 4:41:07 PM

issue #9011 Feature request: Generate images for XML output

Implemented calculation and storage of the resulting images for the XML_OUTPUT and having a reference in the xml file.
Implementation for the the formats `\dotfile`, `\dot`, `\msc`, `\mscfile`, `\startuml` (plantuml), `\diafile` and the formula commands
```
as in the commit:
```
Commit: c47d80d1f810aef9270c1a884099ab0a9a6f8b6d [c47d80d]
Date: Wednesday, January 12, 2022 12:06:10 PM

issue #9011 Feature request: Generate images for XML output

- Introduced setting `XML_INCORPORATE_IMAGES`
- better support for `CREATE_SUBDIRS`
```

it look to me that this has been "factored out".